api/src/wfl/source.clj

```diff
@@ -147,16 +157,37 @@
                                   column dataset)
       (assoc source :dataset dataset))))
 
+(defn combine-tdr-source-details
+  "Reduce to `result` by combining the Terra Data Repo source `_detail`."
+  [result {:keys [datarepo_row_ids end_time snapshot_creation_job_status
+                  start_time] :as _detail}]
+  (if (#{"running" "succeeded"} snapshot_creation_job_status)
+    (-> result
+        (update :datarepo_row_ids into          datarepo_row_ids)
+        (update :end_time         util/latest   end_time)
+        (update :start_time       util/earliest start_time))
+    result))
+
 (defn ^:private find-new-rows
-  "Find new rows in TDR by querying the dataset between the `interval`."
-  [{:keys [dataset table column] :as source}
-   [start end                    :as interval]]
+  "Query TDR for rows within `_interval` that are new to `source`."
+  [{:keys [dataset details table column] :as source}
+   [begin end                            :as _interval]]
   (log/debug (format "%s Looking for rows in %s.%s between [%s, %s]..."
-                     (log-prefix source) (:name dataset) table start end))
-  (-> dataset
-      (datarepo/query-table-between table column interval [:datarepo_row_id])
-      :rows
-      flatten))
+                     (log-prefix source) (:name dataset) table begin end))
+  (let [wfl   (jdbc/with-db-transaction [tx (postgres/wfl-db-config)]
+                (postgres/get-table tx details))
+        old   (when (seq wfl) (reduce combine-tdr-source-details wfl))
+        start (if-let [start_time (:start_time old)]
+                (-> start_time
+                    (util/latest (instant/read-instant-timestamp begin))
+                    timestamp-to-offsetdatetime
+                    (.format bigquery-datetime-format))
+                begin)
+        tdr   (-> dataset
+                  (datarepo/query-table-between
+                   table column [start end] [:datarepo_row_id])
+                  :rows flatten)]
+    (when (seq tdr) (remove (set (:datarepo_row_ids old)) tdr))))
```

Comment:
I'm still not convinced this should be done on every call to `update-source`. I also think the way you're getting the `start_time` is overly complicated - The timestamp when the source was "started" could be written into the database as a new `started` column, rather than setting `last_checked`. The code to compute the start time could then be done once in the loader for any sources that have a `last_checked` but don't have a `started` (i.e. workloads created prior to this change).

Reply:
OK, but WFL still has to read the details to get the row_ids already processed.
Time-fencing the processing interval is trivial aside from the differing formats.
The times are there now. Might as well use them? Why add another timestamp?